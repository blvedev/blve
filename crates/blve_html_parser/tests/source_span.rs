use indoc::indoc;
use insta::assert_debug_snapshot;
use blve_html_parser::{Dom, Result};

#[test]
fn it_can_generate_source_span() -> Result<()> {
    let html = indoc! {"
            <template>
                <h1>Header</h1>
                <p>Paragraph</p>
            </template>
        "};
    let dom = Dom::parse(html)?;
    assert_debug_snapshot!(dom);
    Ok(())
}
