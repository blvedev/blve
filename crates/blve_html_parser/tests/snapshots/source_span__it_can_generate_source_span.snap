---
source: tests/source_span.rs
expression: dom
---
Dom {
    tree_type: DocumentFragment,
    children: [
        Element(
            Element {
                id: None,
                name: "template",
                variant: Normal,
                attributes: {},
                classes: [],
                children: [
                    Element(
                        Element {
                            id: None,
                            name: "h1",
                            variant: Normal,
                            attributes: {},
                            classes: [],
                            children: [
                                Text(
                                    "Header",
                                ),
                            ],
                            source_span: SourceSpan {
                                text: "<h1>Header</h1>",
                                start_line: 2,
                                end_line: 2,
                                start_column: 5,
                                end_column: 20,
                            },
                        },
                    ),
                    Element(
                        Element {
                            id: None,
                            name: "p",
                            variant: Normal,
                            attributes: {},
                            classes: [],
                            children: [
                                Text(
                                    "Paragraph",
                                ),
                            ],
                            source_span: SourceSpan {
                                text: "<p>Paragraph</p>",
                                start_line: 3,
                                end_line: 3,
                                start_column: 5,
                                end_column: 21,
                            },
                        },
                    ),
                ],
                source_span: SourceSpan {
                    text: "<template>\n    <h1>Header</h1>\n    <p>Paragraph</p>\n</template>",
                    start_line: 1,
                    end_line: 4,
                    start_column: 1,
                    end_column: 12,
                },
            },
        ),
    ],
    errors: [],
}
